src/views/settings/Feature/Vault/index.vue

```diff
@@ -62,21 +62,22 @@ export default {
         VAULT_BACKEND: {
           label: this.$t('Provider'),
           hidden: (formValue) => {
-            return !formValue.VAULT_ENABLED
-          }
+            return !formValue.VAULT_ENABLED || formValue['VAULT_BACKEND'] === 'local'
+          },
+          disabled: true
         },
         HCP: {
           label: this.$t('HashicorpVault'),
           component: HashiCorpKV,
-          hidden: (form) => {
-            return form['VAULT_BACKEND'] !== 'hcp'
+          hidden: (formValue) => {
+            return !formValue.VAULT_ENABLED || formValue['VAULT_BACKEND'] !== 'hcp'
           }
         },
         AZURE: {
           label: this.$t('AzureKeyVault'),
           component: AzureKV,
-          hidden: (form) => {
-            return form['VAULT_BACKEND'] !== 'azure'
+          hidden: (formValue) => {
+            return !formValue.VAULT_ENABLED || formValue['VAULT_BACKEND'] !== 'azure'
           }
         }
       },
```

Comment:
The only difference I noticed is the use of "hcp" instead of just "azure". It seems like there was an intentional typo here which should be fixed to avoid any confusion or bugs. The rest of the code doesn't seem problematic based on context provided.

English Response:

No major issues found with the mentioned code changes so far.
Potential improvement could lie in ensuring consistent typographic accuracy for clarity, especially in such critical lines where readability can play a vital role if mistakes occur due to human transcription errors during maintenance processes.
This would be particularly applicable when working closely with legacy systems where maintaining clear coding guidelines might require additional care, given that precision in coding style plays a significant part in system reliability and user experience.
End of suggestion